src/core/SampleClip.cpp

```diff
@@ -129,6 +134,8 @@ void SampleClip::setSampleBuffer(std::shared_ptr<const SampleBuffer> sb)
 	updateLength();
 
 	emit sampleChanged();
+
+	Engine::getSong()->setModified();
```

Comment:
Is this in scope?

Reply:
Same as above with regards to the boyscout rule. Setting the song to modified because something has changed in the core/"business" is not in the responsibility of the presentation/GUI layer. With the changes it will work even if the GUI was switched completely of if there was no GUI and some other entity would listen to song changes.